exams/pearson_test.py

```diff
@@ -0,0 +1,191 @@
+    optional_field,
+    profile_country_to_alpha3
+)
+from profiles.factories import ProfileFactory
+
+
+class PearsonTest(TestCase):
+    """
+    Tests for Pearson code
+    """
+
+    def test_optional_field(self):  # pylint: disable=no-self-use
+        """
+        Test that optional_field returns expected values
+        """
+        with mute_signals(post_save):
+            profile = ProfileFactory.create()
+
+        assert optional_field('address1', profile) == profile.address1
+        assert optional_field('address3', profile) == ''
+
+    def test_get_field_mapper(self):  # pylint: disable=no-self-use
+        """
+        Tests that get_field_mapper handles input correctly
+        """
+        with mute_signals(post_save):
+            profile = ProfileFactory.create()
+
+        def get_addr1(profile):  # pylint: disable=missing-docstring
+            return profile.address1
+
+        assert get_field_mapper('address1')(profile) == profile.address1
+
+        addr1_field_mapper = get_field_mapper(get_addr1)
+
+        assert addr1_field_mapper == get_addr1
+        assert addr1_field_mapper(profile) == profile.address1
+
+        with self.assertRaises(TypeError):
+            get_field_mapper([])
+
+    def test_profile_to_ccd_row_invalid_country(self):  # pylint: disable=no-self-use
+        """
+        A profile with an invalid country code should raise an InvalidProfileDataException
+        """
+        with mute_signals(post_save):
+            profile = ProfileFactory.create()
+        profile.country = 'XXXX'
+        with self.assertRaises(InvalidProfileDataException):
+            profile_country_to_alpha3(profile)
+
+    def test_write_profiles_ccd_no_profiles(self):  # pylint: disable=no-self-use
+        """
+        Tests write_profiles_ccd against an empty set of profiles
+        """
+        file = io.StringIO()
+
+        ccd_writer(file, [])
+
+        lines = file.getvalue().splitlines()
+        header = lines[0].split('\t')
+
+        assert len(lines) == 1
+        assert header == [
+            'ClientCandidateId',
+            'FirstName',
+            'LastName',
+            'Email',
+            'Address1',
+            'Address2',
+            'Address3',
+            'City',
+            'State',
+            'PostalCode',
+            'Country',
+            'Phone',
+            'PhoneCountryCode',
+            'LastUpdate',
+        ]
+
+    def test_ccd_writer(self):  # pylint: disable=no-self-use
+        """
+        Tests ccd_writer against a set of profiles
+        """
+        file = io.StringIO()
+
+        exam_profiles = []
+        with mute_signals(post_save):
+            exam_profiles.append(ExamProfileFactory.create(profile=ProfileFactory.create()))
+            exam_profiles.append(ExamProfileFactory.create(profile=ProfileFactory.create(address3='Room B345')))
+            exam_profiles.append(ExamProfileFactory.create(profile=ProfileFactory.create(address2=None)))
```

Comment:
When you're dealing with Factory Boy factories that use subfactories and related factories, you can set attributed in the dependent factories easily by using double-underscores to separate the relations. For example, you can use this code to accomplish the same thing:

```python
exam_profiles.append(ExamProfileFactory.create())
exam_profiles.append(ExamProfileFactory.create(profile__address3='Room B345'))
exam_profiles.append(ExamProfileFactory.create(profile__address2=None))
```

Then you can reduce the code repetition using `**kwargs` and a list comprehension:
```python
exam_profile_kwargs = [{}, {profile__address3: 'Room B345'}, {profile__address2: None}]
with mute_signals(post_save):
    exam_profiles = [ExamProfileFactory.create(**kwargs) for kwargs in exam_profile_kwargs]
```